packages/cli/package.json

```diff
@@ -25,20 +25,21 @@
     "build": "tsc -b .",
     "docs": "./generate_docs.sh",
     "lint": "tslint -c tslint.json --project tsconfig.json",
-    "prepack": "yarn run build && oclif-dev manifest && oclif-dev readme",
+    "generate:shrinkwrap": "yarn run npm install --production && yarn run npm shrinkwrap",
+    "prepack": "yarn run build && oclif-dev manifest && oclif-dev readme && yarn run generate:shrinkwrap",
     "test:reset": "yarn --cwd ../protocol devchain generate-tar .tmp/devchain.tar.gz --migration_override ../dev-utils/src/migration-override.json --upto 24 --release_gold_contracts scripts/truffle/releaseGoldExampleConfigs.json",
     "test:livechain": "yarn --cwd ../protocol devchain run-tar .tmp/devchain.tar.gz",
     "test": "TZ=UTC jest --runInBand"
   },
   "dependencies": {
-    "@celo/contractkit": "1.0.1-dev",
-    "@celo/explorer": "1.0.1-dev",
-    "@celo/identity": "1.0.1-dev",
-    "@celo/governance": "1.0.1-dev",
-    "@celo/utils": "1.0.1-dev",
-    "@celo/wallet-local": "1.0.1-dev",
-    "@celo/wallet-hsm-azure": "1.0.1-dev",
-    "@celo/wallet-ledger": "1.0.1-dev",
+    "@celo/contractkit": "1.0.1",
+    "@celo/explorer": "1.0.1",
+    "@celo/identity": "1.0.1",
+    "@celo/governance": "1.0.1",
+    "@celo/utils": "1.0.1",
+    "@celo/wallet-local": "1.0.1",
+    "@celo/wallet-hsm-azure": "1.0.1",
+    "@celo/wallet-ledger": "1.0.1",
     "@ledgerhq/hw-transport-node-hid": "~5.11.0",
```

Comment:
Shouldn't we pin this and the oclif packages?

Reply:
ohhhhhhhhh, my bad. I shouldn't push that. It was used to generate the packages that I tested.
Nice catch

Reply:
Oh, you were pointing the other line, sorry (the @celo fixed versions were the wrong ones)

The idea, is that we keep the package.json as the same way we were handling it. We can still upgrade versions or having "non-fixed" ones.
The thing that maintains all the versions, is the `npm-shrinkwrap.json`
That `.json` is the one fixing everything.

(don't know how to reference lines in a comment from a different file)
Take a look on this
This line: https://github.com/celo-org/celo-monorepo/pull/6575/files#diff-25941123ee4dcd8d518962aa40b6b50a4eea5157e5494c135f8efc705fed752cR439
is basically pointing in the shrinkwrap that the library could eventually accept a version increment, but, check this other line:
https://github.com/celo-org/celo-monorepo/pull/6575/files#diff-25941123ee4dcd8d518962aa40b6b50a4eea5157e5494c135f8efc705fed752cR983
That is the line fixing it.

So, in the shrinkwrap file, you have what does every lib requires, but at the bottom, you have the list of the libs that were selected.

Probably, first it generates the file, and then searches for the minimum version acceptable for every lib to work, and then, it fixes that version.

Reply:
ohh i see. So shrinkwrap takes care of pinning the versions, and package.json looks the same.

Reply:
K this makes sense. Thanks for putting up the PR!